### PR TITLE
ROC-2039 Add request default judgment

### PR DESCRIPTION
**Pull request by timja-kainos**
### JIRA link ###
https://tools.hmcts.net/jira/browse/ROC-2039

### CODE REVIEW

File: src/main/app/claims/models/claim.ts [hunk 1]
```diff
@@ -53,4 +53,8 @@ export default class Claim implements Serializable<Claim> {
   get remainingDays (): number {
     return this.responseDeadline.diff(MomentFactory.currentDate(), 'days')
   }
+
+  get eligibleForCCJ (): boolean {
```
Comment (rlewan) [inline src/main/app/claims/models/claim.ts]: Could this be unit tested?

File: src/main/features/ccj/draft/DraftCCJService.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+const deserialize = (value: any): DraftCCJ => {
+  return new DraftCCJ().deserialize(value)
+}
+
+const middleware = new DraftMiddleware<DraftCCJ>('ccj', deserialize)
+
+export class DraftCCJService {
+
+  static retrieve (req: express.Request, res: express.Response, next: express.NextFunction): void {
```
Comment (rlewan) [inline src/main/features/ccj/draft/DraftCCJService.ts]: The name is called retrieve, but it's a `void`. Maybe it would be worth to rename?

Comment (timja-kainos) [inline src/main/features/ccj/draft/DraftCCJService.ts]: Its a pattern throughout the app...

File: src/test/features/ccj/routes/checks/authorization-check.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import * as config from 'config'
+
+import { checkAuthorizationGuards as check } from '../../../../routes/authorization-check'
+
+export const accessDeniedPagePattern = new RegExp(`${config.get('idam.authentication-web.url')}/login\\?continue-url=http://127.0.0.1:[0-9]{1,5}/receiver`)
```
Comment (rlewan) [inline src/test/features/ccj/routes/checks/authorization-check.ts]: This looks like a pattern to a login page, maybe it makes sense to rename?

File: src/test/features/dashboard/routes/claimant.ts [hunk 4]
```diff
@@ -50,6 +52,21 @@ describe('Dashboard - claimant page', () => {
         })
       })
     })
+  })
+  describe('on POST', () => {
```
Comment (rlewan) [inline src/test/features/dashboard/routes/claimant.ts]: This currently reads as:

> on POST when user authorised should redirect to county court judgement flow

Feels like something is missing, maybe `'on request CCJ POST'` would make more sense?

File: src/main/features/ccj/draft/DraftCCJService.ts (hunk 2 above)
Comment (rlewan) [inline src/main/features/ccj/draft/DraftCCJService.ts]: Not a fortunate one though I think. We can do later though, not that important.

File: src/test/features/dashboard/routes/claimant.ts (hunk 4 above)
Comment (timja-kainos) [inline src/test/features/dashboard/routes/claimant.ts]: its actually:
> Dashboard - claimant page on POST when user authorised should redirect to county court judgment flow

File: src/test/features/ccj/routes/checks/authorization-check.ts (hunk 3 above)
Comment (timja-kainos) [inline src/test/features/ccj/routes/checks/authorization-check.ts]: its also the access denied page pattern too.

File: src/main/app/claims/models/claim.ts (hunk 1 above)
Comment (timja-kainos) [inline src/main/app/claims/models/claim.ts]: have pushed a unit test for it

**Pull request merged**